contracts/svm/programs/express_relay/src/lib.rs

```diff
@@ -147,6 +152,155 @@ pub mod express_relay {
             amount,
         )
     }
+
+    pub fn swap(ctx: Context<Swap>, data: SwapArgs) -> Result<()> {
+        validate_ata(
+            &ctx.accounts.trader_input_ata.key(),
+            &ctx.accounts.trader.key(),
+            &ctx.accounts.mint_input.key(),
+        )?;
+        validate_ata(
+            &ctx.accounts.trader_output_ata.key(),
+            &ctx.accounts.trader.key(),
+            &ctx.accounts.mint_output.key(),
+        )?;
+
+        let (input_after_fees, output_after_fees) = match data.fee_token {
+            FeeToken::Input => {
+                validate_ata(
+                    &ctx.accounts.router_fee_receiver_ata.key(),
+                    &ctx.accounts.router.key(),
+                    &ctx.accounts.mint_input.key(),
+                )?;
+
+                validate_ata(
+                    &ctx.accounts.relayer_fee_receiver_ata.key(),
+                    &ctx.accounts.express_relay_metadata.relayer_signer,
+                    &ctx.accounts.mint_input.key(),
+                )?;
+
+                validate_ata(
+                    &ctx.accounts.protocol_fee_receiver_ata.key(),
+                    &ctx.accounts.express_relay_metadata.key(),
+                    &ctx.accounts.mint_input.key(),
+                )?;
+
+                let (remaining_amount, fee_protocol) = perform_fee_split(
+                    data.amount_input,
+                    ctx.accounts.express_relay_metadata.split_protocol * 100,
```

Comment:
we can't use these splits here, they're denominated in terms of the surplus/bid amounts. so they're rather large, e.g. split_router is 50%.

i think those splits in the metadata account should be marked as specifically for Auction Paradigm 1 (i.e. native token surplus bids). For this Auction Paradigm 2 (token transfer fees), we should probably add a new field to Express Relay metadata called like `token_fee`. 

We take `token_fee` bps and split it between the relayer and the express relay (we can probably continue to use the splits we have right now). And then we also take a referral fee that is specified in this instruction and send that to the router. This way, we can add a field to Express Relay Metadata called `token_fee`, and it's fine if it's just set to 0 for now. Wdyt?